lib/galaxy/tools/__init__.py

```diff
@@ -305,13 +305,14 @@ def tools_by_id(self):
         return self._tools_by_id
 
     def get_cache_region(self, tool_cache_data_dir):
-        if tool_cache_data_dir not in self.cache_regions:
-            self.cache_regions[tool_cache_data_dir] = ToolDocumentCache(cache_dir=tool_cache_data_dir)
-        return self.cache_regions[tool_cache_data_dir]
+        if self.app.config.enable_tool_document_cache:
+            if tool_cache_data_dir not in self.cache_regions:
+                self.cache_regions[tool_cache_data_dir] = ToolDocumentCache(cache_dir=tool_cache_data_dir)
+            return self.cache_regions[tool_cache_data_dir]
```

Comment:
This function is now returning nothing if the tool cache is disabled. Should it return None?

Reply:
A function/method that doesn't explicitly return a value will return None. Explicit better than implicit, but shouldn't be a problem.

Reply:
+1 on keeping `return None` implicit. It's a very common case, and it keeps it simple.